core/src/main/java/org/infinispan/commands/VisitableCommand.java

```diff
@@ -10,12 +18,29 @@
  * @author Manik Surtani (<a href="mailto:manik@jboss.org">manik@jboss.org</a>)
  * @since 4.0
  */
-public interface VisitableCommand extends ReplicableCommand {
+public interface VisitableCommand extends CacheRpcCommand {
+
+   static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
 
    default void init(ComponentRegistry registry) {
       // no-op
    }
 
+   default CompletionStage<?> invokeAsync(ComponentRegistry registry) throws Throwable {
+      if (getCacheName() == null)
```

Comment:
can the cache name be null?

Reply:
It's null for `LocalCommand` instances e.g. `GetAllCommand`, but in those cases `invokeAsync` shouldn't be called. We could add `assert getCacheName() != null;` to be sure?